crates/bevy_pbr/src/render/light.rs

```diff
@@ -261,9 +261,9 @@ impl SpecializedMeshPipeline for ShadowPipeline {
 
         Ok(RenderPipelineDescriptor {
             vertex: VertexState {
-                shader: SHADOW_SHADER_HANDLE.typed::<Shader>(),
+                shader: MESH_SHADER_HANDLE.typed::<Shader>(),
                 entry_point: "vertex".into(),
-                shader_defs: vec![],
+                shader_defs: vec!["SHADOW_MAPPING".to_string()],
```

Comment:
For this to be correct in the general case, we need to inherit the shaderdefs of the specialized mesh pipeline in addition to this one.

Reply:
The point was that with SHADOW_MAPPING defined, only code that is either without any other shader defs or that has shader defs that are configured for shadow passes are included.

Reply:
That is a ‘by convention’ maintenance burden, but I think it’s pretty manageable because so little is needed for shadow mapping. Just clip position.

Reply:
I guess your comment was about, for example, VERTEX_TANGENTS. My point is that if SHADOW_MAPPING is set, no other shader defs are part of what remains.

Reply:
This is likely going to be needed when we merge in skinning and morph target support. Where they are needed in the shadow vertex shader to be correct. Any other user made specialized define would also likely need to be reflected here too.

Reply:
Yeah, I realised what you meant when I looked over the mesh skinning PR last night. I'll have to figure out how to hook that up then.